vault/login_mfa.go

```diff
@@ -107,7 +107,7 @@ func (b *SystemBackend) loginMFAPaths() []*framework.Path {
 }
 
 func genericOptionalUUIDRegex(name string) string {
-	return fmt.Sprintf("(/(?P<%s>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}))?", name)
+	return fmt.Sprintf("(/(?P<%s>[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}))?", name)
```

Comment:
I'm not sure this would be of practical benefit for users, because the only UUIDs that need to be accepted here are ones which were generated by Vault - so this would only be to accomodate users who receive a UUID from Vault and then decide to convert it to uppercase before returning it to the Vault API... I'm not sure that's something that needs to be catered for.